src/PageContent/Calendar/DescriptionContent.js

```diff
@@ -92,6 +111,18 @@ class DescriptionContent extends Component {
         });
     }
 
+    saveNotes(e) {
+        e.preventDefault();
+        this.setState({ waiting: true });
+        let updates = {};
+        updates['notes'] = this.state.currentNote;
+        deliveriesRef.child(this.props.delivery.id).update(updates).then(() => {
+            // record timestamp of when the write was done
+            this.setState({ savedTimestamp: moment().valueOf() });
+        });
+        this.setState({ waiting: false, isEditingNotes: false });    
```

Comment:
This should only become false after the update is complete. This is some weird firebase async stuff, so we can talk about this in person.